src/core/server/cross_compatibility/cross_compatibility_service.ts

```diff
@@ -0,0 +1,115 @@
+import { Logger } from '../logging';
+import { OpenSearchServiceStart } from '../opensearch';
+import { CompatibleEnginePluginVersions, PluginName } from '../plugins/types';
+
+export interface StartDeps {
+  opensearch: OpenSearchServiceStart;
+  plugins: Map<PluginName, CompatibleEnginePluginVersions>;
+}
+
+export class CrossCompatibilityService {
+  private readonly log: Logger;
+
+  constructor(coreContext: CoreContext) {
+    this.log = coreContext.logger.get('version-compatibility-service');
+  }
+
+  start({ opensearch, plugins }: StartDeps): CrossCompatibilityServiceStart {
+    this.log.warn('Starting version compatibility service');
+    return {
+      verifyOpenSearchPluginsState: (pluginName: string) => {
+        const pluginOpenSearchDeps = plugins.get(pluginName) || {};
+        return this.verifyOpenSearchPluginsState(opensearch, pluginOpenSearchDeps);
+      },
+    };
+  }
+
+  public async getOpenSearchPlugins(opensearch: OpenSearchServiceStart) {
+    // Makes cat.plugin api call to fetch list of OpenSearch plugins installed on the cluster
+    try {
+      const { body } = await opensearch.client.asInternalUser.cat.plugins<any[]>({
+        format: 'JSON',
+      });
+      return body;
+    } catch (error) {
+      this.log.warn(
+        `Cat API call to OpenSearch to get list of plugins installed on the cluster has failed: ${error}`
+      );
+      return [];
+    }
+  }
+
+  public async checkPluginVersionCompatibility(
+    pluginOpenSearchDeps: CompatibleEnginePluginVersions,
+    opensearchInstalledPlugins: CatPluginsResponse
+  ) {
+    const results: CrossCompatibilityResult[] = [];
+    for (const [pluginName, versionRange] of Object.entries(pluginOpenSearchDeps)) {
+      // add check to see if the Dashboards plugin version is compatible with installed OpenSearch plugin
+      const {
+        isCompatible,
+        installedPluginVersions,
+      } = await this.isVersionCompatibleOSPluginInstalled(
+        opensearchInstalledPlugins,
+        pluginName,
+        versionRange
+      );
+      results.push({
+        pluginName,
+        isCompatible: !isCompatible ? false : true,
+        incompatibleReason: !isCompatible
+          ? `OpenSearch plugin "${pluginName}" in the version range "${versionRange}" is not installed on the OpenSearch for the OpenSearch Dashboards plugin to function as expected.`
+          : '',
+        installedVersions: installedPluginVersions,
+      });
+
+      if (!isCompatible) {
+        this.log.warn(
+          `OpenSearch plugin "${pluginName}" is not installed on the cluster for the OpenSearch Dashboards plugin to function as expected.`
+        );
+      }
+    }
+    return results;
+  }
+
+  private async verifyOpenSearchPluginsState(
+    opensearch: OpenSearchServiceStart,
+    pluginOpenSearchDeps: CompatibleEnginePluginVersions
+  ): Promise<CrossCompatibilityResult[]> {
+    this.log.warn('Checking OpenSearch Plugin version compatibility');
```

Comment:
Why `WARN`?

Reply:
You're right! This is duplicated, instead this can be an `info`.